internal/pkgmanager/ecosystem.go

```diff
@@ -28,6 +28,12 @@ var (
 	}
 )
 
+/*
+defaultArchiveFilename returns a naive default choice of filename from a
+download URL by simply returning everything after the final slash in the URL.
+There is no guarantee that this method results in a good archive filename - e.g.
+NPM package namespace is not always included in the URL-derived filename.
+*/
```

Comment:
Long term this could be improved in two ways:

(1) inspecting the content-disposition header in the response for the filename flag
(2) parsing the URL and extracting the path component

The result may still not be good though.